homeassistant/components/recorder/statistics.py

```diff
@@ -239,6 +239,18 @@ def _configured_unit(unit: str, units: UnitSystem) -> str:
     return unit
 
 
+def clear_statistics(hass: HomeAssistant, statistic_ids: list[str]) -> None:
+    """Clear statistics for a list of statistic_ids."""
+    with session_scope(hass=hass) as session:
+        metadata_ids = _get_metadata_ids(hass, session, statistic_ids)
+        session.query(Statistics).filter(
+            Statistics.metadata_id.in_(metadata_ids)
+        ).delete(synchronize_session=False)
+        session.query(StatisticsMeta).filter(
+            StatisticsMeta.id.in_(metadata_ids)
+        ).delete(synchronize_session=False)
```

Comment:
I think there's a race condition here if hourly statistic generation is running in parallel with this

Reply:
Yes, you should create a statistics mutation lock or insert the cleaning as a task into the recorder task queue.

Reply:
> insert the cleaning as a task into the recorder task queue

That sounds like the better option, with no risk for any deadlocks etc.
Would it be enough OK to mention in the docstring for the WS API that the wiping does not happen immediately, so the call just returns after posting the job to the recorder queue?

Reply:
Yeah that's fine.